### PR TITLE
Add new filter types and rework filter screen layout

### PR DESCRIPTION
**Pull request by maurizi**
- General refactor of `FilterDisplay` and `BaseFilter` to push rendering logic from `FilterDisplay` into each `BaseFilter` subclass.
 - Adds a `DateRangeFilter` by making it a subclass of the now abstract `RangeFilter` and moving some existing logic into a new `NumericRangeFilter`
 - Adds a new`TextFilter` class

Connects to #239
Connects to #240 
Connects to #251

### CODE REVIEW

**Comment by maurizi** [comment]
@azavea-bot rebuild

**Comment by hectcastro** [comment]
@azavea-bot rebuild

File: OpenTreeMap/src/main/java/org/azavea/helpers/DateButtonListener.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            final DatePickerDialog dialog = new DatePickerDialog(activity, (v, year, month, day) -> {
+                final String updatedTimestamp = getTimestamp(activity, year, month, day);
+                final String displayDate = formatTimestampForDisplay(updatedTimestamp);
+
+                button.setText(displayDate);
+                button.setTag(tag, updatedTimestamp);
+            }, cal.get(Calendar.YEAR), cal.get(Calendar.MONTH), cal.get(Calendar.DAY_OF_MONTH));
+            if (setTimestamp != null) {
+                dialog.setButton(DatePickerDialog.BUTTON_NEUTRAL, activity.getString(R.string.date_field_clear), (d, which) -> {
+                    button.setTag(tag, null);
```
Comment (maurizi) [inline OpenTreeMap/src/main/java/org/azavea/helpers/DateButtonListener.java]: The clear button is working on the plot detail screen, but not the filter screen - need to investigate

**Comment by maurizi** [comment]
When loading the filter screen with enough filters that it scrolls, the scroll area is vertically centered - it should be at the top

File: OpenTreeMap/src/main/java/org/azavea/helpers/DateButtonListener.java (hunk 1 above)
Comment (steventlamb) [inline OpenTreeMap/src/main/java/org/azavea/helpers/DateButtonListener.java]: could not reproduce

**Comment by steventlamb** [comment]
+1, tested, refactor looks good

**Pull request merged**